### ISSUE TITLE
override_options cpp_std does not work for vs backend

### ISSUE THREAD

**Issue by DFOVIT** [mesonbuild/meson#10393]
**Describe the bug**
override_options cpp_std is broken with the vs backend while it is ok with the ninja backend. It fails to override 

**To Reproduce**
main.cpp:
```c++
#include <any>

int main(int, char *[]) {
    std::any ret = 0;
    return std::any_cast<int>(ret);
}
```
meson.build :
```
project('vsbackend-override-options', 'cpp',
    version : '1.0',
    default_options : ['cpp_std=c++11',])

executable('override-options', 'main.cpp', override_options: 'cpp_std=c++17')
```

```shell
meson --vsenv --backend=vs2017 builddir
meson compile -C builddir
```

**Expected behavior**
Compiles with vs2017 (and others) backend.

**system parameters**
* plain native build
* Windows 10
* Python 3.10.3
* Meson 0.62.1

**Comment by marzer** [comment]
Came here to report this. Also happens for the `c_std` option.

**Comment by marzer** [comment]
Actually, it's worse than that, it turns out:

In C++ this causes MSVC to emit a harmess-but-annoying message, e.g.:
```
cl : command line  warning D9025: overriding '/std:c++20' with '/std:c++14'
```

But their C frontend handles this scenario differently, instead emitting a hard error:
```
cl : command line  error D8016: '/std:c11' and '/std:c17' command-line options are incompatible
```

Meaning that testing scenarios like "test that my library's public API is consumable in all these versions of C" are now much more difficult/complicated since they require multiple invocations `meson setup`.

@eli-schwartz Any chance a fix for this can make 0.64.0?
